Games/core/Game.js

```diff
@@ -791,12 +794,48 @@ module.exports = class Game {
         this.processAlertQueue();
         this.events.emit("meetingsMade");
 
+        this.vegMeeting = undefined;
+
         // Create next state timer
         this.createNextStateTimer(stateInfo);
     }
 
     createNextStateTimer(stateInfo) {
-        this.createTimer("main", stateInfo.length, () => this.gotoNextState());
+        this.createTimer("main", stateInfo.length, () => this.checkVeg());
+        this.checkAllMeetingsReady();
+    }
+
+    checkVeg() {
+        this.clearTimer("main");
+        this.clearTimer("secondary");
+        this.vegMeeting = this.createMeeting(VegReadyMeeting, "vegKickMeeting");
+        var vegMeetings = Object.values(this.history.states[this.currentState].meetings).filter(x => x.id !== this.vegMeeting.id);
+        for (let player of this.players) {
+            var addToKick = true;
+            if (player.alive) {
+                for (let i = 0; i < vegMeetings.length; i++) {
+                    if (vegMeetings[i].members[player.id] !== undefined) {
+                        if (vegMeetings[i].votes[player.id] === undefined) {
+                            if (vegMeetings[i].members[player.id].canVote && vegMeetings[i].members[player.id].canUpdateVote) {
+                                addToKick = false;
+                            }
+                        }
+                    }
+                }
+            }
+            else {
+                addToKick = false;
+            }
+            if (addToKick) {
+                this.vegMeeting.join(player);
+            }
+        }
+
+        this.vegMeeting.init();

```

Comment:
this part is the same as instant meeting logic, might combine next time but not important to do that